modules/mobile/app/services/mobile/v2/appointments/proxy.rb

```diff
@@ -37,26 +37,32 @@ def get_appointments(start_date:, end_date:, include_pending:, pagination_params
 
         def merge_clinic_facility_address(appointments)
           cached_facilities = {}
+
+          facility_ids = appointments.map(&:location_id).compact.uniq
+          facility_ids.each do |facility_id|
+            cached_facilities[facility_id] = appointments_helper.get_facility(facility_id)
+          end
+
           appointments.each do |appt|
             facility_id = appt[:location_id]
             next unless facility_id
 
-            cached = cached_facilities[facility_id]
-            cached_facilities[facility_id] = appointments_helper.get_facility(facility_id) unless cached
-
             appt[:location] = cached_facilities[facility_id]
           end
         end
 
         def merge_auxiliary_clinic_info(appointments)
           cached_clinics = {}
+
+          location_clinics = appointments.map { |appt| [appt.location_id, appt.clinic] }.reject { |a| a.any?(nil) }.uniq
```

Comment:
It's worth noting that this is a slight divergence from the existing code in that the existing code only skips if the clinic id is nil, not if the location id is nil. The service method doesn't make it clear if it can work without location id, but i suspect it can't and this is correct.

Reply:
I can change it if you'd like, it read to me like a nil location id would result in nil but I'm not knowledgable here

Reply:
I think what you have is more correct. I'm just pointing out that it's a slight change. I've never noticed errors around this call but it would be easy to miss them in sentry. This probably makes the code better.

Reply:
Also, I see that we're doing it elsewhere in this test, but I still believe that 
`expect(response.parsed_body['location']).to be_nil`
is just incorrect. You can also find this within this test:
`location = response.parsed_body.dig('data', 0, 'attributes', 'location')`
I believe that's the correct way to do it. Can you please fix that mistake where you find it?

Reply:
Ok just updated the spec that checks for location to be nil and I'm actually testing the object structure now